contracts-periphery/test/UniswapWithdrawHook.fork.1.t.sol

```diff
@@ -0,0 +1,173 @@
+  uint256 numOfAddrs = 10;
+  address sponsor;
+  uint256 quote;
+  uint256 feeAmount;
+
+  // Mainnet Addresses
+  address public constant dai = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+  address public constant weth = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+  address public constant router = 0x68b3465833fb72A70ecDF485E0e4C7bD8665Fc45;
+  address public constant quoter = 0xb27308f9F90D607463bb33eA1BeBb41C27CE5AB6;
+  // 0.3% pool fee is unique to this DAI-WETH pool
+  uint24 poolFee = 3000;
+
+  // Umbra functions that we'll be calling
+  enum UmbraFns {withdrawTokenAndCall, withdrawTokenAndCallOnBehalf}
+
+  function setUp() public override {
+    super.setUp();
+    umbraContract = IUmbra(umbra);
+    swapRouter = ISwapRouter(router);
+    quoterContract = IQuoter(quoter);
+    withdrawHook = new UniswapWithdrawHook(ISwapRouter(swapRouter));
+    daiToken = IERC20(dai);
+    // Alice's address is derived from the private key of 1.
+    alice = vm.addr(1);
+    // Owner approves tokens
+    withdrawHook.approveToken(daiToken);
+    deal(dai, address(this), 10e29);
+  }
+
+  function testFuzz_HookTest_withdrawTokenAndCall(
+    uint256 amount,
+    uint256 swapAmount,
+    uint256 feeBips,
+    uint256 sponsorFee
+  ) public {
+    // Using for loop here to generate deterministic addresses instead of fetching random address state
+    // from network RPC node every time a fuzz test is run
+    for (uint256 i = 0; i < numOfAddrs; i++) {
+      feeRecipient = address(uint160(uint256(keccak256(abi.encode(i)))));
+      recipient = address(uint160(uint256(keccak256(abi.encode(feeRecipient)))));
+      _testFuzz_HookTest(
+        recipient,
+        amount,
+        swapAmount,
+        feeBips,
+        feeRecipient,
+        sponsorFee,
+        UmbraFns.withdrawTokenAndCall
+      );
+    }
+  }
+
+  function testFuzz_HookTest_withdrawTokenAndCallOnBehalf(
+    uint256 amount,
+    uint256 swapAmount,
+    uint256 feeBips,
+    uint256 sponsorFee
+  ) public {
+    // Using for loop here to generate deterministic addresses instead of fetching random address state
+    // from network RPC node every time a fuzz test is run
+    for (uint256 i = 0; i < numOfAddrs; i++) {
+      feeRecipient = address(uint160(uint256(keccak256(abi.encode(i)))));
+      recipient = address(uint160(uint256(keccak256(abi.encode(feeRecipient)))));
+      _testFuzz_HookTest(
+        recipient,
+        amount,
+        swapAmount,
+        feeBips,
+        feeRecipient,
+        sponsorFee,
+        UmbraFns.withdrawTokenAndCallOnBehalf
+      );
+    }
+  }
+
+  function _testFuzz_HookTest(
+    address destinationAddr,
+    uint256 amount,
+    uint256 swapAmount,
+    uint256 feeBips,
+    address feeReceiver,
+    uint256 sponsorFee,
+    UmbraFns fn
+  ) public {
+    amount = bound(amount, 10e10, 100000 ether);
+    swapAmount = bound(swapAmount, 10e8, amount);
+    feeBips = bound(feeBips, 1, 100);
+    sponsorFee = bound(sponsorFee, 1, 1000);
```

Comment:
Taking a fresh look at this, I can't quite recreate the issue I was having. Thinking of changing the bound for `sponsorFee` to 
```suggestion
    sponsorFee = bound(sponsorFee, 1, amount - swapAmount);
```
Which seem to pass the test with 10k runs.

Reply:
Basically we're saying is:
- Send `amount` with Umbra
- Of that, the recipient swaps `swapAmount` of it to ETH
- This leaves a max fee for the sponsor of `amount - swapAmount`

Makes sense to me!